js/sign/src/signers/node-crypto-signing-strategy.ts

```diff
@@ -1,12 +1,12 @@
 import crypto, { KeyObject } from 'crypto';
-import { checkIsValidEd25519Key } from '../utils/utils.js';
+import { checkIsValidKey } from '../utils/utils.js';
 import { ISigningStrategy } from './signing-strategy-interface.js';
 
 // Class to be used when signing with parsed `crypto.KeyObject` private key
 // provided directly in the constructor.
 export class NodeCryptoSigningStrategy implements ISigningStrategy {
   constructor(private readonly privateKey: KeyObject) {
-    checkIsValidEd25519Key('private', privateKey);
+    checkIsValidKey('private', privateKey);
   }
 
   async sign(data: Uint8Array): Promise<Uint8Array> {
```

Comment:
The main question here -- in your opinion, does it make sense to state the algorithm explicitly for clarity in `.sign()` and `.verify()` or omit it for simplicity?

[crypto](https://nodejs.org/api/crypto.html#cryptosignalgorithm-data-key-callback) documentation states that missing algorithm will be inferred based on the key type; I've validated that assumption and it turns out that for ECDSA P-256 keys the hashing algorithm chosen is indeed SHA-256.

Reply:
I don't have a strong opinion here. Maybe if kept like this, could add a function comment on where it's inferred from

Reply:
I don't have a strong opinion but would be fine to keep it as is.

Reply:
Added a comment then.